File: 1сем/ОАиП лабы/Common7/IDE/ProjectTemplatesCache/VisualBasic/WCF/1049/WcfServiceLibrary/IService1.vb
' ПРИМЕЧАНИЕ. Можно использовать команду "Переименовать" в контекстном меню, чтобы изменить имя интерфейса "IService1" в коде и файле конфигурации.
<ServiceContract()>$if$ ($targetframeworkversion$ <= 3.5) _$endif$
Public Interface IService1

    <OperationContract()>$if$ ($targetframeworkversion$ <= 3.5) _$endif$
    Function GetData(ByVal value As Integer) As String

    <OperationContract()>$if$ ($targetframeworkversion$ <= 3.5) _$endif$
    Function GetDataUsingDataContract(ByVal composite As CompositeType) As CompositeType

    ' TODO: Добавьте здесь операции служб

End Interface

' Используйте контракт данных, как показано на следующем примере, чтобы добавить сложные типы к сервисным операциям.
$if$ ($targetframeworkversion$ >= 4.5)' В проект можно добавлять XSD-файлы. После построения проекта, вы можете напрямую использовать в нем определенные типы данных с пространством имен "$safeprojectname$.ContractType".
$endif$$if$ ($targetframeworkversion$ <= 3.5)
<DataContract()> _
Public Class CompositeType

    Private boolValueField As Boolean
    Private stringValueField As String

    <DataMember()> _
    Public Property BoolValue() As Boolean
        Get
            Return Me.boolValueField
        End Get
        Set(ByVal value As Boolean)
            Me.boolValueField = value
        End Set
    End Property

    <DataMember()> _
    Public Property StringValue() As String
        Get
            Return Me.stringValueField
        End Get
        Set(ByVal value As String)
            Me.stringValueField = value
        End Set
    End Property
$else$
<DataContract()>
Public Class CompositeType

    <DataMember()>
    Public Property BoolValue() As Boolean

    <DataMember()>
    Public Property StringValue() As String
$endif$
End Class

File: 1сем/ОАиП лабы/Common7/IDE/ProjectTemplates/CSharp/WCF/1049/WcfServiceLibrary/IService1.cs
using System;
using System.Collections.Generic;
$if$ ($targetframeworkversion$ >= 3.5)using System.Linq;
$endif$using System.Runtime.Serialization;
using System.ServiceModel;
using System.Text;

namespace $safeprojectname$
{
	// ПРИМЕЧАНИЕ. Команду "Переименовать" в меню "Рефакторинг" можно использовать для одновременного изменения имени интерфейса "IService1" в коде и файле конфигурации.
	[ServiceContract]
	public interface IService1
	{
		[OperationContract]
		string GetData(int value);

		[OperationContract]
		CompositeType GetDataUsingDataContract(CompositeType composite);

		// TODO: Добавьте здесь операции служб
	}

	// Используйте контракт данных, как показано на следующем примере, чтобы добавить сложные типы к сервисным операциям.
	$if$ ($targetframeworkversion$ >= 4.5)// В проект можно добавлять XSD-файлы. После построения проекта вы можете напрямую использовать в нем определенные типы данных с пространством имен "$safeprojectname$.ContractType".
	$endif$[DataContract]
	public class CompositeType
	{
		bool boolValue = true;
		string stringValue = "Hello ";

		[DataMember]
		public bool BoolValue
		{
			get { return boolValue;}
			set { boolValue = value;}
		}

		[DataMember]
		public string StringValue
		{
			get { return stringValue;}
			set { stringValue = value;}
		}
	}
}
